Runtime/World/Entity.h

```diff
@@ -127,28 +129,16 @@ namespace Spartan
 			VALIDATE_COMPONENT_TYPE(T);
 			const ComponentType type = IComponent::TypeToEnum<T>();
 
-			std::vector<std::shared_ptr<T>> components;
-			for (const auto& component : m_components)
-			{
-				if (component->GetType() != type)
-					continue;
-
-				components.emplace_back(std::static_pointer_cast<T>(component));
-			}
+            auto world = m_context->GetSubsystem<World>();
+            auto components = world->GetComponentManager<T>()->GetComponents(GetId());
 
-			return components;
+            return components;
 		}
 		
 		// Checks if a component of ComponentType exists
 		bool HasComponent(const ComponentType type) 
 		{ 
-			for (const auto& component : m_components)
-			{
-				if (component->GetType() == type)
-					return true;
-			}
-
-			return false;
+            return m_component_mask.mask & (1 << (int)type);
```

Comment:
ComponentMask has AddComponent() and RemoveComponent() functions, however here, HasComponent() is done inside the Entity instead.
It would be more clear if all were exposed in the Entity or all were provided by the ComponentMask, including HasComponent().
However, an ideal solution would be to delete ComponentMask, and simply have an integer inside Entity.

Reply:
Thanks for the comment. I removed the ComponentMask class and now all the calculations are done in the Entity class